src/components/general/MarketOrderAction/index.tsx

```diff
@@ -20,57 +18,44 @@ export type MarketOrderActionProps = {
   setPrice: any
   setAmount: any
   account: any
+  blockchainApi: any
+  orderType: string
 }
-const MarketOrderAction = ({ type = 'Buy', setOpenOrder, price, amount, setPrice, setAmount, account }: MarketOrderActionProps) => {
-  // const wsProvider = new WsProvider('wss://blockchain.polkadex.trade:9955');
-  const wsProviderInstance = webSocket;
-
+const MarketOrderAction = ({ type = 'Buy', setOpenOrder, price, amount, setPrice, setAmount, account, blockchainApi, orderType }: MarketOrderActionProps) => {
 
   const [slider, setSlider] = useState({ values: [50] })
-  const [available, setAvailable] = useState(2)
+  const [available, setAvailable] = useState(0)
 
   const tradingPairID = "0xf28a3c76161b8d5723b6b8b092695f418037c747faa2ad8bc33d8871f720aac9";
   const UNIT = 1000000000000;
 
   useEffect(() => {
-    const fetchAvailableBalance = async () => {
-      // const api = await ApiPromise.create({ provider: wsProvider });
-      // Wait until we are ready and connected
-      // await api.isReady;
+    blockchainApi?.query.genericAsset.freeBalance(type === 'Buy' ? 1 : 2, account.address, (data) => {
+      setAvailable(+data.toString() / UNIT);
+    });
+  }, [blockchainApi])
 
-      // Do something
-      // console.log(api.genesisHash.toHex());
-      // api.query.genericAsset.FreeBalance(type === 'Buy' ? 1 : 2, account.address, (data) => {
-      //   console.log('Market Order');
-      //   console.log(data);
-      // });
+  const cleanString = (value) => {
+    let pos = value.indexOf(".");
+    if (pos === -1 ){
+      return value;
+    } else {
+      return value.substring(0, pos);
     }
-    fetchAvailableBalance()
-  }, [])
+  }
 
   const startTransaction = async () => {
     if (account.address) {
-      const api = await ApiPromise.create({ provider: wsProviderInstance });
       const polkadotExtensionDapp = await import('@polkadot/extension-dapp');
       const injector = await polkadotExtensionDapp.web3FromSource(account.meta.source);
-      let transferExtrinsic;
-      if (type === 'Buy') {
-        toast.success('Buy initiated');
-        transferExtrinsic = api.tx.polkadex.submitOrder(
-          "BidLimit",
-          tradingPairID,
-          (parseFloat(price + '') * UNIT),
-          (parseFloat(amount + '') * UNIT)
-        );
-      } else if (type === 'Sell') {
-        toast.success('Sold initiated');
-        transferExtrinsic = api.tx.polkadex.submitOrder(
-          "AskLimit",
-          tradingPairID,
-          (parseFloat(price + '') * UNIT),
-          (parseFloat(amount + '') * UNIT)
-        );
-      }
+
+      toast.success(type + ' initiated');
+      let transferExtrinsic = blockchainApi.tx.polkadex.submitOrder(
+        type === 'Buy' ? "BidLimit" : "AskLimit",
```

Comment:
@RudarDaman is this used for both market and limit? if yes, then this is wrong.

1. Dropdown: Limit Order, Buy: BidLimit
2. Dropdown: Limit Order , Sell: AskLimit 
3. Dropdown: Market Order , Sell: BidMarket
4. Dropdown: Market Order , Sell: AskMarket